packages/backend/src/open_payments/quote/model.ts

```diff
@@ -0,0 +1,171 @@
+import assert from 'assert'
+import { Model, Pojo } from 'objection'
+import * as Pay from '@interledger/pay'
+
+import { Amount, AmountJSON } from '../amount'
+import { Account } from '../account/model'
+import { Asset } from '../../asset/model'
+import { BaseModel } from '../../shared/baseModel'
+
+export class Quote extends BaseModel {
+  public static readonly tableName = 'quotes'
+
+  static get virtualAttributes(): string[] {
+    return [
+      'sendAmount',
+      'receiveAmount',
+      'minExchangeRate',
+      'lowEstimatedExchangeRate',
+      'highEstimatedExchangeRate'
+    ]
+  }
+
+  // Open payments account id of the sender
+  public accountId!: string
+  public account?: Account
+
+  // Asset id of the sender
+  public assetId!: string
+  public asset!: Asset
+
+  static relationMappings = {
+    account: {
+      relation: Model.HasOneRelation,
+      modelClass: Account,
+      join: {
+        from: 'quotes.accountId',
+        to: 'accounts.id'
+      }
+    },
+    asset: {
+      relation: Model.HasOneRelation,
+      modelClass: Asset,
+      join: {
+        from: 'quotes.assetId',
+        to: 'assets.id'
+      }
+    }
+  }
+
+  public expiresAt!: Date
+
+  public receivingPayment!: string
+  public completeReceivingPayment!: boolean
```

Comment:
Why is that part of the quote?

Reply:
This is used to track the case where
>Outgoing payment will transition `receivingPayment` to completed if the Incoming Payment was created (without an `incomingAmount`) by the quote.

https://github.com/interledger/rafiki/pull/326/files#diff-e2d479f40419be082f7f09ef1c4e8243e9dca4c023baea84c412615ac20df4c3R155-R156
https://github.com/interledger/rafiki/pull/326/files#diff-b3cdd360f857e6c7414751f228aa27f2287a899ca10737d16726bed355e16bdeR148

Reply:
So why not store it in the outgoing payment?

Reply:
It's a detail that needs to tracked from when the quote is created (before the outgoing payment is created) to when the outgoing payment is completed.
We could duplicate it to the outgoing payment, but most of the outgoing payment details are use the fields of the fetched corresponding quote.